src/core/store.ts

```diff
@@ -26,20 +26,65 @@ type Revision = number
 type InvalidatedRevision = number
 type ReadDependencies = Map<AnyAtom, Revision>
 
-// immutable atom state
+/**
+ * Immutable atom state, tracked for both mounted and unmounted atoms in a store.
+ */
 export type AtomState<Value = unknown> = {
+  /**
+   * Counts number of times atom has actually changed or recomputed.
+   */
   r: Revision
+  /**
+   * Marks the revision of this atom when a transitive dependency was invalidated.
+   * Mounted atoms are considered invalid when `r === i`. Unmounted atoms are
+   * always considered invalid.
+   */
   i?: InvalidatedRevision
+  /**
+   * Maps from a dependency to the dependency's revision when it was last read.
+   * We can skip recomputation of an atom by comparing the ReadDependencies revision
+   * of each dependency to that dependencies's current revision.
+   */
   d: ReadDependencies
 } & ({ e: ReadError } | { p: SuspensePromise } | { v: ResolveType<Value> })
 
-export type VersionObject = { p?: VersionObject } // "p"arent version
+/**
+ * Represents a version of a store. A version contains a state for all atoms.
+ *
+ * In concurrent React rendering, state can "branch" during transitions: the
+ * current state may continue to be rendered while a new state is being built
+ * concurrently. We key our atom state containers by this global version to
+ * represent the state for each diverging branch.
+ *
+ * While a new version is being built, we read atom previous state from the
+ * previous version.
+ */
+export type VersionObject = {
+  /**
+   * "p"arent version.
+   *
+   * Once a version is committed completely, the `p` property is deleted so the
+   * child version is independent, and the parent revision can be garbage
+   * collected.
+   *
+   * See [Provider] for more details on version data flow.
+   */
+  p?: VersionObject
+}
 
 type Listeners = Set<(version?: VersionObject) => void>
 type Dependents = Set<AnyAtom>
+
+/**
+ * State tracked for atoms with subscribers, or transitive dependencies of atoms
+ * with subscribers. It is freed once an atom no longer has subscribers.
```

Comment:
There are mounted atoms without subscribers. They are only dependencies of other atoms.